src/plugins/flow.js

```diff
@@ -1315,6 +1318,22 @@ export default (superClass: Class<Parser>): Class<Parser> =>
       }
     }
 
+    shouldParseExportStar(): boolean {
+      return (
+        super.shouldParseExportStar() ||
+        (this.isContextual("type") && this.lookahead().type === tt.star)
+      );
+    }
+
+    parseExportStar(node: N.ExportNamedDeclaration, allowNamed: boolean): void {
```

Comment:
Here it should really be `node: N.ExportAllDeclaration`, but it would require some refactoring to `statements.js#parseExport()`